### ISSUE TITLE
Panic on v1.13.2

### ISSUE THREAD

**Issue by jose-ledesma** [aws/amazon-vpc-cni-k8s#2448]
**What happened**:
VPC CNI panic

**Attach logs**
```
$ kubectl logs aws-node-4pts8 -p
Installed /host/opt/cni/bin/aws-cni
Installed /host/opt/cni/bin/egress-cni
time="2023-06-29T15:12:21Z" level=info msg="Starting IPAM daemon... "
time="2023-06-29T15:12:21Z" level=info msg="Checking for IPAM connectivity... "
time="2023-06-29T15:12:23Z" level=info msg="Copying config file... "
time="2023-06-29T15:12:23Z" level=info msg="Successfully copied CNI plugin binary and config file."
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x560ff5791b21]

goroutine 398 [running]:
github.com/aws/amazon-vpc-cni-k8s/pkg/awsutils.(*EC2InstanceMetadataCache).DescribeAllENIs(0xc0009508c0)
	/go/src/github.com/aws/amazon-vpc-cni-k8s/pkg/awsutils/awsutils.go:1239 +0xd01
github.com/aws/amazon-vpc-cni-k8s/pkg/ipamd.(*IPAMContext).nodeIPPoolReconcile(0xc000950780, {0x560ff6815970, 0xc000134020}, 0xdf8475800)
	/go/src/github.com/aws/amazon-vpc-cni-k8s/pkg/ipamd/ipamd.go:1412 +0x5e9
github.com/aws/amazon-vpc-cni-k8s/pkg/ipamd.(*IPAMContext).StartNodeIPPoolManager(0xc000950780)
	/go/src/github.com/aws/amazon-vpc-cni-k8s/pkg/ipamd/ipamd.go:699 +0x66
created by main._main
	/go/src/github.com/aws/amazon-vpc-cni-k8s/cmd/aws-k8s-agent/main.go:68 +0x42b
time="2023-06-29T15:12:28Z" level=error msg="Failed to wait for IPAM daemon to complete" error="exit status 2"
```

**What you expected to happen**:
`aws-node` pods don't panic ^^

**How to reproduce it (as minimally and precisely as possible)**:
unsure

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Server Version: version.Info{Major:"1", Minor:"23+", GitVersion:"v1.23.17-eks-c12679a", GitCommit:"d5ce2cee85d99653d6f8c278043213db21b1cd72", GitTreeState:"clean", BuildDate:"2023-05-22T20:32:28Z", GoVersion:"go1.19.6", Compiler:"gc", Platform:"linux/amd64"}
```
- CNI Version: v.1.13.2
- OS (e.g: `cat /etc/os-release`):
```
$ cat /etc/os-release
NAME="Amazon Linux"
VERSION="2"
ID="amzn"
ID_LIKE="centos rhel fedora"
VERSION_ID="2"
PRETTY_NAME="Amazon Linux 2"
ANSI_COLOR="0;33"
CPE_NAME="cpe:2.3:o:amazon:amazon_linux:2"
HOME_URL="https://amazonlinux.com/"
```
- Kernel (e.g. `uname -a`):
```
$ uname -a
Linux ip-xx-xx-xx-xx.xxxxx.compute.internal 5.4.242-156.349.amzn2.x86_64 #1 SMP Tue May 23 18:48:04 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by jdn5126** [comment]
@jose-ledesma what instance type is the node seeing this issue? This looks somewhat familiar to me in that it could happen if the instance type has multiple network cards, which VPC CNI does not support

**Comment by pcn** [comment]
You've seen it fail with a SIGSEGV? I would expect a runtime failure or other out of bounds error from the go runtime, unless it's passing something weird into the libc or other non-go dependency.

**Comment by jdn5126** [comment]
I haven't seen a SIGSEGV, as we would certainly fix any crash we saw. I am going off of the line this points to: https://github.com/aws/amazon-vpc-cni-k8s/blob/v1.13.2/pkg/awsutils/awsutils.go#L1239 and trying to come up with a theory. It would help to get the `/var/log/aws-routed-eni/ipamd.log` output as well

**Comment by jose-ledesma** [comment]
> @jose-ledesma what instance type is the node seeing this issue? This looks somewhat familiar to me in that it could happen if the instance type has multiple network cards, which VPC CNI does not support

This happened both in a `m5.xlarge` and a `c6i.8xlarge` (I just updated the CNI today, across ~22 nodes in 2 different EKS clusters). For context, we are installing the VPC CNI as an EKS add-on, and we have not seen this behaviour in version `v1.11.4` 

> It would help to get the /var/log/aws-routed-eni/ipamd.log output as well

Let me grab and review the log before sharing it publicly

**Comment by jdn5126** [comment]
@jose-ledesma Feel free to email `k8s-awscni-triage@amazon.com` if that is easier. If you take that route, the full node logs would be best (from `sudo bash /opt/cni/bin/aws-cni-support.sh`). It would also be good to include how you are deploying the CNI, i.e. EKS addon, manifest from GitHub, or Helm chart

**Comment by jose-ledesma** [comment]
> @jose-ledesma Feel free to email `k8s-awscni-triage@amazon.com` if that is easier. If you take that route, the full node logs would be best (from `sudo bash /opt/cni/bin/aws-cni-support.sh`). It would also be good to include how you are deploying the CNI, i.e. EKS addon, manifest from GitHub, or Helm chart

The nodes have been already terminated, but I can still grab the logs for specific files from our logs platform (so, let me know if there are any other log useful)

**Comment by jose-ledesma** [comment]
> @jose-ledesma Feel free to email `k8s-awscni-triage@amazon.com`

just sent the log file 

> It would also be good to include how you are deploying the CNI, i.e. EKS addon, manifest from GitHub, or Helm chart

it has been installed as an EKS Addon (and we didn't have panics with 1.11.4, the issue has started today after upgrading to 1.13.2)

**Comment by jdn5126** [comment]
@jose-ledesma any update or resolution here?

**Comment by mraerino** [comment]
has there been a release that is expected to fix this crash?

**Comment by jdn5126** [comment]
The provided logs did not contain any crash, and this is not reproducible. The log line in question that this links to cannot crash from any method I attempted

**Comment by jdn5126** [comment]
Closing as fixed by #2467
